### ISSUE TITLE
Fresh install of enterprise_extensions errors out with circular import

### ISSUE THREAD

**Issue by golamshaifullah** [nanograv/enterprise_extensions#234]
As the title says, importing extensions fails. Tested with both containers and conda environment builds.  

```
Successfully installed enterprise-pulsar-3.3.3 enterprise_extensions-2.4.3 healpy-1.16.6
❯ ipython --pylab
Python 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:50:21) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.
Using matplotlib backend: QtAgg

In [1]: import enterprise_extensions as e

In [2]: e.__version__
Out[2]: '2.4.3'

In [3]: from enterprise_extensions import blocks
 ---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[3], line 1
----> 1 from enterprise_extensions import blocks

File /work/git_projects/enterprise_extensions_ng/enterprise_extensions/blocks.py:18
     16 from . import gp_kernels as gpk
     17 from . import model_orfs
---> 18 from . import model_utils
     21 __all__ = [
     22     "white_noise_block",
     23     "red_noise_block",
   (...)
     28     "common_red_noise_block",
     29 ]
     32 def channelized_backends(backend_flags):

File /work/git_projects/enterprise_extensions_ng/enterprise_extensions/model_utils.py:14
     11 except ImportError:
     12     from emcee.autocorr import integrated_time as acor
---> 14 from enterprise_extensions import models
     16 # Log-spaced frequncies
     19 def linBinning(T, logmode, f_min, nlin, nlog):

File /work/git_projects/enterprise_extensions_ng/enterprise_extensions/models.py:16
     14 from enterprise_extensions import dropout as do
     15 from enterprise_extensions import model_utils
---> 16 from enterprise_extensions.blocks import (bwm_block, bwm_sglpsr_block,
     17                                           chromatic_noise_block,
     18                                           common_red_noise_block,
     19                                           dm_noise_block, red_noise_block,
     20                                           white_noise_block)
     21 from enterprise_extensions.chromatic.solar_wind import solar_wind_block
     22 from enterprise_extensions.timing import timing_block

ImportError: cannot import name 'bwm_block' from partially initialized module 'enterprise_extensions.blocks' (most likely due to a circular import) (/work/git_projects/enterprise_extensions_ng/enterprise_extensions/blocks.py) 
```

**Comment by nobleson-phy** [comment]
The issue is resolved for me if I import models using 'from enterprise_extensions import models' before importing blocks.

**Comment by golamshaifullah** [comment]
@nobleson-phy - indeed we can hack scripts to work, but this does break old scripts. The import line in model_utils is never used, so I would still propose removing it, as in the submitted PR #235.

**Comment by siyuan-chen** [comment]
Hi all,
the main problem here is that `model.py` uses `model_utils.py` and `model_utils.py` uses `models.py`. So, you get the circular import issue.
Models.py uses model_utils to get the time span and other functions to set the frequency binning of the power law model. This has been fine in the past, I think the `CompareTimingModel()` code was recently added to model_utils and reads the models from models.py. So, this part causes the issue.

There are several solutions and I think we should have a broad discussion which one we want to go for:
1. remove `import model_utils` from `models.py. This would require to set frequencies and time spans in each model separately without using functions from model_utils. (Or have those functions moved to a different file or into models.py)
2. remove `import models` from model_utils.py. This would mean to move `CompareTimingModel()` out of model_utils. And a) make it a separate file, b) put it into models.py or c) put it somewhere else (avoiding a circular import)

**Comment by siyuan-chen** [comment]
@golamshaifullah , I think think your PR will not be a complete fix. `CompareTimingModel()` needs models.py to work and there should be more place in models, where model_utils is called. (These will be added through the big merge between the EPTA and the bcyweek branch).

**Comment by achalumeau** [comment]
@golamshaifullah , @siyuan-chen , @vhaasteren, I just submitted a new PR that (I think) solves the entire issue: https://github.com/nanograv/enterprise_extensions/pull/243.
It follows the option 2 from Siyuan. I have placed the class CompareTimingModel in timing.py, as it looks like the most logical to me.

**Comment by AaronDJohnson** [comment]
This should be fixed with #244
